src/js/models/messages.js

```diff
@@ -78,12 +82,40 @@ export default class Messages{
 		try{
 			await db.collection("Profiles").where("name", "==", searchContent).get().then(function(querySnapshot) {
 				querySnapshot.forEach(function(doc) {
-					arr.push(doc.data());
+					arr.push({
+						data:doc.data(),
+						id: doc.id
+					});
 				});
 			});
 			this.searchRes = arr;
 		} catch(error){
 			alert(error);
 		}
 	}
+
+	async doesChatExists(targetUid,selfUid){
+		try{
+			const chats = await db.collection("Messages").where("chatterIds","array-contains",selfUid).get();
+			chats.forEach(chat => {
+				if(chat.data().chatterIds[0] == targetUid || chat.data().chatterIds[1] == targetUid){
+					console.log(chat.data());
+					this.chatExists = true;
+				}
+			});
+		} catch(error){
+			alert(error);
+		}
+	}
+
+	async createNewChat(chatterId,chatter,myName){
```

Comment:
Rather than passing in `name`, why not store `name` in the [constructor](https://github.com/Yuxuan-Lin/VioLegacy/pull/33/files#diff-8ff12f9af9c29ec28325c2225dfa9adfa12627d678c2db8a31f9c382ece8af29R3) like `uid`

Reply:
We will change that ASAP.